### PR TITLE
Favor pathlib objects instead of str(fp)

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
Simplifications possible now that 3.6 is the minimum Python version.

Also, `Environment.getTempFile()` and `_EnvironmentCore.getTempFile()` default `returnPathlib=True`.

Also, updated path to Preview on failing tests in environment.py's `testToSettings()`

### CODE REVIEW

File: music21/environment.py [hunk 1]
```diff
@@ -1462,7 +1465,7 @@ def testToSettings(self):
   <preference name="braillePath" />
   <preference name="debug" value="0" />
   <preference name="directoryScratch" />
-  <preference name="graphicsPath" value="/Applications/Preview.app" />
+  <preference name="graphicsPath" value="/System/Applications/Preview.app" />
```
Comment (ahankinson) [inline music21/environment.py:1468]: I don't think this is the correct path for macOS Preview.app

Comment (ahankinson) [reply music21/environment.py:1468]: Actually, nevermind. I'm completely wrong. I hadn't noticed that Apple changed this around. Sorry.

Comment (jacobtylerwalls) [reply music21/environment.py:1468]: No problem! In 6.1 there is a new function to derive the path to Preview based on the macOS version, but these tests had not been updated, so they would happen to pass on Travis CI but fail on my Mac.

**Comment by mscuthbert** [comment]
Had a chance to review this and it looks great!

**Comment by mscuthbert** [comment]
Thanks for finding all the 3.6 references.  V.7 is likely to be Python 3.7+ only, though I'm not so dying to use dataclasses or other Py3.7 features except `from __future__ import annotations`, so music21 v.7 might still work (unofficially) on Py3.6.  When we get to Py 3.8 as the minimum though, the breakage will happen fast, since there are a lot of features of Py3.8 that I really love esp. for Typing (TypedDict!).  But that won't happen for another 2 years.

**Pull request merged**